app/response_processor.py

```diff
@@ -46,41 +52,44 @@ def service_name(self, url=None):
             elif 'validate' in parts:
                 return 'SDX-VALIDATE'
         except AttributeError as e:
-            self.logger.error(e)
+            self.logger.error("No valid service name", exception=e)
 
-    def process(self, encrypted_survey):
-        # decrypt
+    def _process(self, encrypted_survey):
         decrypted_json = self.decrypt_survey(encrypted_survey)
 
         metadata = decrypted_json['metadata']
-        self.logger = self.logger.bind(user_id=metadata['user_id'], ru_ref=metadata['ru_ref'])
+        self.logger = self.logger.bind(user_id=metadata['user_id'],
```

Comment:
Use the `get` method on all the dictionary access calls here to avoid `KeyError` exceptions when a malformed message is passed.